programs/mmm/src/state.rs

```diff
@@ -78,6 +78,9 @@ pub struct Pool {
     pub payment_mint: Pubkey,
     pub allowlists: [Allowlist; ALLOWLIST_MAX_LEN],
     pub buyside_payment_amount: u64,
+
+    pub shared_escrow_account: Option<Pubkey>,
+    pub shared_escrow_cap: Option<u64>,
```

Comment:
any chance we can just not have these as options? I think we can change `using_shared_escrow` to check if the pubkey is equal to `Pubkey::default()`

Reply:
Also, I think the `shared_escrow_cap` can just be a counter, no need to tie it to the payment amount, could indexing a bit easier

Reply:
curious what's the benefit of not having these fields as optional?

Reply:
I guess it's to avoid unwraping etc

Reply:
I'll fix these, had some access issue when pushing to this repo

Reply:
I feel store the shared_escrow_cap in SOL would be easier for our api to determine wether this pool need more funds, since `buyside_payment_amount` will be 0 in this case

Reply:
Two things

- If this pool needs more funds cannot be derived from the cap, it needs to compare what's in the shared_escrow_account's lamports balance, and the required fulfillment spot_price (i.e. the total price to be fulfilled)

- To determine is a pool is valid:

1. expired ?
2. shared_escrow_count > 0 ?
3. shared_escrow_account.lamports() > total_price ?